scipy/stats/_morestats.py

```diff
@@ -942,25 +942,34 @@ def boxcox_llf(lmb, data):
     >>> plt.show()
 
     """
-    data = np.asarray(data)
+    xp = array_namespace(data)
+    data = xp.asarray(data)
     N = data.shape[0]
     if N == 0:
-        return np.nan
+        return xp.nan
 
-    logdata = np.log(data)
+    if xp.isdtype(data.dtype, 'integral'):
+        data = xp.asarray(data, dtype=xp.float64)
+    
+    logdata = xp.log(data)
 
     # Compute the variance of the transformed data.
     if lmb == 0:
-        logvar = np.log(np.var(logdata, axis=0))
+        logvar = xp.log(xp.var(logdata, axis=0))
```

Comment:
Separately, we'll want to add `axis` support. Would you be willing to do that?

Reply:
I can try!